### DISCUSSION TITLE
v1.98.0

### DISCUSSION THREAD

**Discussion by cyring** [cyring/CoreFreq#500] [Show and tell]
# [Build]
* Change dkms configuration to install into `/usr/bin/`
* Solved compilation with Fedora using `RHEL MINOR` version number `99`
* [CR] Fixed some compilation warnings
* [CR] Fix memory allocation in kernel pages for the SysGate
* Workaround to `musl` change in `basename` (#494)
* aarch64: Build with Redhat RHEL version 9
* [CR] x86_64: Prevent array size overflow
* `dkms` and `ckms` configuration files upgraded to version `1.98`

# [Intel]
* [Raptor Lake] Adding the HB DID of i7-14700K' IMC
* Mitigation mechanism: GDS_NO; RFDS_NO; MONITOR_MITG_NO
* Apply some SDM Documentation Changes
* [MTL] Fix CMD Stretch bit range width
* Added method `CLOCK_FLEX_MAX` with Xeon's Nehalem & Core 2
    * Grants full `MSR_FLEX_RATIO` access to tested architectures:
    - Alder Lake/S
    - Tiger Lake/U
    - Westmere/Gulftown
* Provides the overclocking bins with unlocked processors
* Query the Overclocking bit (OC) from Capabilities
* Adding technologies: VMD, HDCP, IPU and VPU
    * Volume Management Device
    * High-Bandwidth Digital Content Protection
    * Image Processing Unit
    * Vision Processing Unit
* Gaussian & Neural Accelerator - GNA technology
* [CR] Renamed Gemini Lake structures
* [Meteor Lake] Introducing memory controller decoder

# [AArch64]
* Optimizing the Vcore seek
* Workaround to Package discrete voltage: the highest Vcore
* Improving computation of Euclidean division
* Adjust frequency division in `CNTFRQ` and `PMU` counters
* Scale a frequency factor from the interval
* Refactoring the frequency ratios for decimal precision
* Compile `dev_pm_opp_put` if Kernel greater or equal `4.11`
* Get the voltage core of CPUs from OPP

# [AMD]
* [Zen4][Raphael] Adding the EPYC 4004 Series
* [Zen4] Clarify Hawk Point and Phoenix-Refresh architectures
* [Family 19h] New voltage formula assigned to model 61h
    * Vcore activated
    * Voltage SoC deactivated
* [Zen] Prevent the calculation of negative temperature (issue #496)
* [Family 18h] Documented Hygon C86 7375
* [Family 19h] PStateDef specification: Adding `VID[8]` bit 32
    * Specification of MSR `HW_PSTATE_STATUS`

# [CLI]
* aarch64: Display the state of the Memory Management Unit (MMU)
* aarch64: State of Instruction Cache Unit `I$` and Data Cache Unit `D$`
* Added `UI_RULER_MINIMUM` & `UI_RULER_MAXIMUM` building constraints
* Revert "[CLI] Responsive ruler to architectural context"
    * This reverts commit 4bc38d696fa739d7fcd7a8f542b8069a93bcbbc4.
* x86_64: AMD Boost and P-States redesigned
* [UI] Hardening missing console/terminal size
* Responsive ruler to architectural context
* x86_64: Make HDCP meaning string shorter

# [Driver]
* Created `C2U_Enable` as a parameter alias of `C1U_Enable`

<hr /><em>This discussion was created from the release <a href='https://github.com/cyring/CoreFreq/releases/tag/1.98.0'>v1.98.0</a>.</em>

**Comment by cyring** [comment]
@bubbleguuum @inglor @ich777

Hello,

This release candidate fixes a buffer overflow in `corefreqk.ko` detected by kernel 6.x

I highly recommend to upgrade the distribution packages

Greetings
CyrIng

**Reply by bubbleguuum** [reply]
Will do this week-end.

**Reply by ich777** [reply]
The package for the stable unRAID release was already built about six hours ago and I just triggered the build for the beta version, should be done in a few moments. Thanks for the update!

**Comment by cyring** [comment]
By the way, you can use the patch from issue #498 if your setup is Linux-Next based

**Comment by bubbleguuum** [comment]
openSUSE TW now updated to v1.98.0.

**Reply by cyring** [reply]
Thank you very much.

**Comment by cyring** [comment]
@bubbleguuum @inglor @ich777 Hello,

Commit 70a527c5cdf170fb73becc3d27c482ae4903a22b aims to pretty print the build output

```
$ make -j
  CC [build/corefreqd.o]
  CC [build/corefreqm.o]
  CC [build/corefreq-cli.o]
  CC [build/corefreq-ui.o]
  CC [build/corefreq-cli-rsc.o]
  CC [build/corefreq-cli-json.o]
  CC [build/corefreq-cli-extra.o]
  LD [build/corefreqd]
  LD [build/corefreq-cli]
  CC [M]  /home/username/src/CoreFreq/build/module/corefreqk.o
  LD [M]  /home/username/src/CoreFreq/build/corefreqk.o
  MODPOST /home/username/src/CoreFreq/build/Module.symvers
  CC [M]  /home/username/src/CoreFreq/build/corefreqk.mod.o
  LD [M]  /home/username/src/CoreFreq/build/corefreqk.ko

$ make -j clean
  CLEAN [M] /home/username/src/CoreFreq/build
  CLEAN [build/corefreqd]
  CLEAN [build/corefreq-cli]
  CLEAN [build/module/corefreqk.c]
  CLEAN [build/Makefile]
  RMDIR [build/module]
  RMDIR [build]
```
In addition to that is introduced the verbose level `V` which is borrowed from the kernel `Makefile`
```
$ make help
o---------------------------------------------------------------o
|  make [all] [clean] [info] [help] [install] [module-install]  |
|                                                               |
|  V=<n>                                                        |
|    where <n> is the kernel verbose build level                |
|                                                               |
```

Remarks
* `make V=0` is going to produce the similar outputs of previous CoreFreq
* `make V=1` is `V=0` plus the detailed log of the kernel module build

Please let me know if those changes don't _hurt_ your package ?
And feel free to suggest enhancements of the kind.

Regards
Cyril

**Reply by ich777** [reply]
I assume if you pass over nothing it will default to `V=0` correct, if that‘s the case that will be fine for my build process.

**Reply by cyring** [reply]
No, if you pass over nothing, it will default to the new pretty print.
You have to explicit add `V=0` for _old style_ log
